bin/check-puppet-last-run.rb

```diff
@@ -95,7 +107,11 @@ def run
       @message += " with #{@failures} failures"
     end
 
-    if @now - @last_run > config[:crit_age] || @failures > 0
+    if @restart_failures > 0
+      @message += " with #{@restart_failures} restart failures"
+    end
+
+    if @now - @last_run > config[:crit_age] || @failures > 0 || @restart_failures > 0
```

Comment:
hmm this should probably be `if @now - @last_run > config[:crit_age] || @failures > 0 || (@restart_failures > 0 && config[report_restart_failures:])`